04 Strategy Library/999 Seasonality and Session Effect in Index Futures/03 Method.html

```diff
@@ -0,0 +1,124 @@
+<h3>Step 1:  Initialization</h3>
+
+<p>
+	We set up an 8-year long backtest period from 2008-01-01 to 2015-12-31. Then, we choose the E-mini S&P 500 futures, sort them in ascending order according to days to expiration and pick the nearest expiration contract. After that, we set the Bullish Season and the Bearish Season based on months and set the Night Session based on time of a day. We trigger the open and the close of trades with self.Schedule.On() function.
+</p>
+
+<div class="section-example-container">
+
+<pre class="python">
+	self.SetStartDate(2008, 1, 1)  # Set Start Date
+    self.SetEndDate(2015, 12, 31) # Set End Date
+    self.SetCash(100000)  # Set Strategy Cash
+        
+    future = self.AddFuture(Futures.Indices.SP500EMini) # S&P E-mini future contracts
+    future.SetFilter(timedelta(0), timedelta(180)) # Filter according to Expiry date
+    
+    # Set benchmark
+    benchmark = self.AddEquity("SPY")
+    self.SetBenchmark(benchmark.Symbol)
+    
+    # Set season
+    # Bullish Season: Nov - May; 
+    # Bearish Season: Jun - Oct
+    
+    # self.bullSeason = {11, 12, 1, 2, 3, 4, 5} # Bullish Season
+    
+    self.bearSeason = {6, 7, 8, 9, 10} # Bearish Season
+    
+    # Overnight trading strategy
+    
+    
+    # Bullish Season: buy at 18:00, sell at 5:30; buy at 13:15, sell at 15:15
+    # Bearlish Season: buy at 18:00, sell at 5:30; buy at 13:15, sell at 14:00
+    
+    
+    # Night: 1800 - 530
+    # Afternoon: 1315 - 1400
+    # Close: 1400 - 1515
+    
+    # Night session
+    self.Schedule.On(self.DateRules.EveryDay('SPY'), self.TimeRules.At(18, 0), Action(self.Enter))
+    self.Schedule.On(self.DateRules.EveryDay('SPY'), self.TimeRules.At(5, 30), Action(self.Exit))
+    
+    # Afternoon session
+    self.Schedule.On(self.DateRules.EveryDay('SPY'), self.TimeRules.At(13, 15), Action(self.Enter))
+    self.Schedule.On(self.DateRules.EveryDay('SPY'), self.TimeRules.At(14, 0), Action(self.ExitException))
+    
+    # Close session
+    self.Schedule.On(self.DateRules.EveryDay('SPY'), self.TimeRules.At(14, 0), Action(self.EnterException))
+    self.Schedule.On(self.DateRules.EveryDay('SPY'), self.TimeRules.At(15, 15), Action(self.Exit))
+
+</pre>
+
+</div>
+
+<h3>Step 2:  Set up the Trading Signals</h3>
+
+<p>
+	For Combined Strategy, in Bullish Season, we buy the contract at 13:15, sell at 15:15 ( Afternoon Session + Close Session) and then buy again at 18:00 and sell it at 5:30 the next day (Night Session); in Bearish Season, we buy the contract at 13:15, sell at 14:00 ( Afternoon Session only) and then buy again at 18:00 and sell it at 5:30 the next day (Night Session). So there are two trades for every trading day. To realize this, we just need to check the month of the trading time and judge when to close the first trade (at 15:15 or at 14:00). 
```

Comment:
Change 5:30 to 05:30. a lot of our users are American and won't immediately realize we say 5:30 to mean 5:30AM. Using 05:30 helps convey this and the fact that we are using a 24-hr clock in the write-up